### PR TITLE
Update documentation on usage with Nuke

### PR DESCRIPTION
**Pull request by scottmerchant**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,13 +38,24 @@ You should see output that looks something like this:
...
+```
+
+Add a parameter marked with `[OctoVersion]` and it will be auto-populated:
 
 ```c#
-[NukeOctoVersion] readonly OctoVersionInfo OctoVersionInfo;
+// The Required Attribute will automatically throw an exception if the 
+// OctoVersionInfo parameter is not set due to an error or misconfiguration in Nuke.
+[Required]
+[OctoVersion(AutoDetectBranchName = true)]
```
Comment (matt-richardson) [inline README.md:53]: We don't want to recommend this - see [this part](https://github.com/OctopusDeploy/OctoVersion#cant-you-figure-out-the-branch-im-on-why-do-i-have-to-specify-it) of the readme.

The approach I've used in a bunch of places is:
```
    [Parameter("Branch name for OctoVersion to use to calculate the version number. Can be set via the environment variable OCTOVERSION_CurrentBranch.", Name = "OCTOVERSION_CurrentBranch")] readonly string BranchName;
    [Parameter("Whether to auto-detect the branch name - this is okay for a local build, but should not be used under CI.")] readonly bool AutoDetectBranch = IsLocalBuild;
    [OctoVersion(UpdateBuildNumber = true, BranchParameter = nameof(BranchName), AutoDetectBranchParameter = nameof(AutoDetectBranch), Framework = "net6.0")] readonly OctoVersionInfo OctoVersionInfo;
```

(the `Framework = "net6.0"` is only required if your app is net6)

**Review by matt-richardson — CHANGES_REQUESTED** [review]
Thanks for this - appreciate it.
We don't really want to recommend `AutoDetectBranchName` though...

File: README.md (hunk 1 above)
Comment (scottmerchant) [reply README.md]: Thanks for that, yeah I wasn't sure how many/which properties to show for the example. I've changed it to your suggestion. Does it look ok?

**Review by matt-richardson — APPROVED** [review]
Nice one - works for me.

The real question is "would this have helped *you* succeed"?

**Comment by scottmerchant** [comment]
> Nice one - works for me.
> 
> The real question is "would this have helped _you_ succeed"?

I think so! especially because the example now matches the usage in Octopus Server. And the step about adding the package with the nuke command.

**Comment by ryangribble** [comment]
Just a comment about
> "Can be set via the environment variable OCTOVERSION_CurrentBranch.", 

Whilst this is true for OctoVersion itself, if this is a documented Nuke build example wouldnt it be better to specify it using the nuke parameter that the comment is referring to.  in other words, this is a nuke parameter so it can be set by passing `--BranchName` to nuke, or by using the env variable that nuke would associate with that parameter?

**Comment by matt-richardson** [comment]
> Just a comment about
> 
> > "Can be set via the environment variable OCTOVERSION_CurrentBranch.",
> 
> Whilst this is true for OctoVersion itself, if this is a documented Nuke build example wouldnt it be better to specify it using the nuke parameter that the comment is referring to. in other words, this is a nuke parameter so it can be set by passing `--BranchName` to nuke, or by using the env variable that nuke would associate with that parameter?
> 
> It kinda doesnt make sense to wire this all up in Nuke but then use OctoVersion's direct env var to set/override the parameters for octoversion 🤔

That's a good point. In this case, we are setting the name of the `BranchName` parameter to `OCTOVERSION_CurrentBranch`, as a shortcut, so we can keep using the existing env vars we have setup... 🤔

**Comment by scottmerchant** [comment]
> > Just a comment about
> > > "Can be set via the environment variable OCTOVERSION_CurrentBranch.",
> > 
> > 
> > Whilst this is true for OctoVersion itself, if this is a documented Nuke build example wouldnt it be better to specify it using the nuke parameter that the comment is referring to. in other words, this is a nuke parameter so it can be set by passing `--BranchName` to nuke, or by using the env variable that nuke would associate with that parameter?
> > It kinda doesnt make sense to wire this all up in Nuke but then use OctoVersion's direct env var to set/override the parameters for octoversion 🤔
> 
> That's a good point. In this case, we are setting the name of the `BranchName` parameter to `OCTOVERSION_CurrentBranch`, as a shortcut, so we can keep using the existing env vars we have setup... 🤔

Do you want me to change this example to just use a CurrentBranch parameter which isn't sourced by an Env Var?

**Comment by matt-richardson** [comment]
> Do you want me to change this example to just use a CurrentBranch parameter which isn't sourced by an Env Var?

I think we can change it to
```
[Parameter("Branch name for OctoVersion to use to calculate the version number. ")] 
readonly string BranchName;
```

The "match env var to parameter" is native to nuke, so this probably matches standard usage better.

**Comment by scottmerchant** [comment]
New PR here for subsequent change: https://github.com/OctopusDeploy/OctoVersion/pull/63

**Pull request merged**